src/types/timezone.jl

```diff
@@ -30,6 +31,37 @@ function _reload_cache!(cache::AbstractDict, compiled_dir::AbstractString)
     return cache
 end
 
+const _TZ_CACHE_INITIALIZED = Threads.Atomic{Bool}(false)
+function initialize_tz_cache()
+    # Write out our compiled tzdata representations into a scratchspace
+   desired_version = TZData.tzdata_version()
+
+   _COMPILED_DIR[] = if desired_version == TZJData.TZDATA_VERSION
+       TZJData.ARTIFACT_DIR
+   else
+       # @info "Loading tzdata $desired_version"
+       TZData.build(desired_version, _scratch_dir())
+   end
+
+   # Load the pre-computed TZData into memory.
+   _reload_cache(_COMPILED_DIR[])
+   _TZ_CACHE_INITIALIZED[] = true
```

Comment:
I would either set this in `get_tz_cache` or move the locking logic into this function